### PR TITLE
refactor (zkquiz): use aligned sdk

### LINKED ISSUE #539: refactor (zkquiz): use aligned sdk
Refactor ZKQuiz to use Aligned SDK.

### PR DESCRIPTION
**Pull request by taturosati**
ZK quiz now uses aligned sdk. This simplifies the flow.

To test follow instructions on examples/zkquiz/README.md copied below

# ZK Aligned Quiz

## Requirements

1. [Rust](https://www.rust-lang.org/tools/install)
2. [Foundry](https://getfoundry.sh)

### 1 - Create Keystore

You can use cast to create a local keystore.
If you already have one you can skip this step.

```bash
cast wallet new-mnemonic
```

Then you can import your created keystore using:

```bash
cast wallet import --interactive <path_to_keystore.json>
```

Then you need to obtain some funds to pay for gas and proof verification.
You can do this by using this [faucet](https://cloud.google.com/application/web3/faucet/ethereum/holesky)

### 2 - Answer Quiz

To answer quiz questions run:

```bash
make answer_quiz KEYSTORE_PATH=<path_to_keystore.json>
```

This will:

1. Ask quiz questions
2. Generate ZK proof
3. Pay & submit proof to aligned for verification
4. Wait for proof to be verified in aligned
5. Claim NFT if proof is verified

### CODE REVIEW

File: examples/zkquiz/quiz/script/src/main.rs [hunk 1]
```diff
@@ -33,21 +81,99 @@ fn main() {
...
+            //
+            //         println!("Proof verified in Aligned, claiming prize...");
+            //
+            //         if let Err(e) = verify_batch_inclusion(
+            //             aligned_verification_data.clone(),
+            //             signer.clone(),
+            //             args.verifier_contract_address,
+            //         ).await {
+            //             println!("Failed to claim prize: {:?}", e);
+            //         }
```
Comment (entropidelic) [inline examples/zkquiz/quiz/script/src/main.rs]: should this code be removed?

Comment (taturosati) [reply examples/zkquiz/quiz/script/src/main.rs]: fixed

**Review by NicolasRampoldi — APPROVED** [review]

File: examples/zkquiz/Makefile [hunk 2]
```diff
@@ -1,39 +1,12 @@
...
-    --proving_system SP1 \
-    --proof quiz/script/proof-with-io.json \
-    --vm_program quiz/program/elf/riscv32im-succinct-zkvm-elf \
-    --proof_generator_addr $(ADDRESS) \
-    --conn wss://batcher.alignedlayer.com
+	@. ./contracts/.env && . ./contracts/deploy.sh
 
 CONTRACT_ADDRESS=0x8dB9e6f1393c3486F30181d606312ec632189621
-RPC_URL=https://ethereum-holesky-rpc.publicnode.com
+RPC_URL=https://rpc.alignedlayer.com
```
Comment (JuArce) [inline examples/zkquiz/Makefile]: This RPC will be shut down . Why has it changed from the public one?

File: examples/zkquiz/quiz/script/src/main.rs [hunk 3]
```diff
@@ -1,16 +1,64 @@
...
+    VerifierContract,
+    "VerifierContract.json",
+);
+
+const BATCHER_URL: &str = "wss://batcher.alignedlayer.com";
+const BATCHER_PAYMENTS_ADDRESS: &str = "0x815aeCA64a974297942D2Bbf034ABEe22a38A003";
 const ELF: &[u8] = include_bytes!("../../program/elf/riscv32im-succinct-zkvm-elf");
 
-fn main() {
+/// Simple program to greet a person
```
Comment (JuArce) [inline examples/zkquiz/quiz/script/src/main.rs]: I think this comment is outdated from an example

File: examples/zkquiz/quiz/script/src/main.rs [hunk 4]
```diff
@@ -1,16 +1,64 @@
...
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    /// Name of the person to greet
```
Comment (JuArce) [inline examples/zkquiz/quiz/script/src/main.rs]: I think this comment is outdated from an example

**Review by JuArce — CHANGES_REQUESTED** [review]

File: examples/zkquiz/Makefile (hunk 2 above)
Comment (taturosati) [reply examples/zkquiz/Makefile]: changed back

File: examples/zkquiz/quiz/script/src/main.rs (hunk 3 above)
Comment (taturosati) [reply examples/zkquiz/quiz/script/src/main.rs]: removed

File: examples/zkquiz/quiz/script/src/main.rs (hunk 4 above)
Comment (taturosati) [reply examples/zkquiz/quiz/script/src/main.rs]: removed

**Review by JuArce — COMMENTED** [review]
lgtm

**Review by entropidelic — APPROVED** [review]

**Review by JuArce — APPROVED** [review]

**Pull request merged**